caproto/ioc_examples/gamepad.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+"""
+This example requires a gamepad and the Python library evdev.
+
+This only works on linux (as it uses the linux kernel input events).
+"""
+from caproto.server import pvproperty, PVGroup, template_arg_parser, run
+from evdev import InputDevice
+
+
+class GampadIOC(PVGroup):
+    def __init__(self, event_path, **kwargs):
+        super().__init__(**kwargs)
+        self._event_path = event_path
+
+    # normal buttons
+    a = pvproperty(value=0, max_length=1)
+    b = pvproperty(value=0, max_length=1)
+    x = pvproperty(value=0, max_length=1)
+    y = pvproperty(value=0, max_length=1)
+    lb = pvproperty(value=0, max_length=1)
+    rb = pvproperty(value=0, max_length=1)
+
+    # joysticks
+
+    # left joystick
+    lx = pvproperty(value=0, max_length=1)
+    ly = pvproperty(value=0, max_length=1)
+    ld = pvproperty(value=0, max_length=1)
+
+    # right joystick, always analog
+    rx = pvproperty(value=0, max_length=1)
```

Comment:
`max_length` is assumed 1 if you pass `pvproperty` a scalar

Also, if this is an analog, it should be `0.0` instead of `0` to get the logic to pick `float` instead of `int`.

Reply:
It is analog, but it seems to be ints in the range [-32k, 32k] vs the 'digital' which seem to be [-1, 0, 1]

Reply:
Thinking a bit more about this - unless we can be certain that these inputs will always be 16-bit (which is -32k to 32k), it'd be preferable to normalize those values somehow.